src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DoNotIgnoreReturnValue.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    using static MicrosoftNetCoreAnalyzersResources;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotIgnoreReturnValueAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string CA2022RuleId = "CA2022";
+
+        internal static readonly DiagnosticDescriptor DoNotIgnoreReturnValueRule = DiagnosticDescriptorHelper.Create(
+            CA2022RuleId,
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueTitle)),
+            CreateLocalizableResourceString(nameof(DoNotIgnoreReturnValueMessage)),
+            DiagnosticCategory.Reliability,
+            RuleLevel.IdeSuggestion,
```

Comment:
If we really hold the line on the attribute only being applied when not using the return value is always (or 99.9% of the time) a bug, we could consider making this a warning by default.

Reply:
(nodding) I figured we could revise this later after we get it integrated into dotnet/runtime and get some APIs annotated and see how many hits we get across the dotnet org repos.

Do you agree that we'd revisit the level later, or should we lean into Warning now?

Reply:
Whichever you prefer.  But if we're going to revisit the level, I think we should do it before shipping; otherwise, folks implicitly take a dependency on the levels we defaulted to and if we increase the severity that can cause headaches.

Reply:
OK. I'll go ahead and upgrade to warning now then. And if we see too many adverse effects when rolling it out through the dotnet org repos, we can dial it back. But I do agree that the value of the analyzer is much higher when it's warning by default.